internal/workload/manager.go

```diff
@@ -554,3 +557,29 @@ func (w *WorkloadManager) podAuthModified(authPath string, auth string) bool {
 	}
 	return bytes.Compare(file, []byte(auth)) != 0
 }
+
+func (w *WorkloadManager) updateSecrets(configuration models.DeviceConfigurationMessage) []error {
+	deviceSecrets, err := w.workloads.ListSecrets()
+	if err != nil {
+		return []error{err}
+	}
+	var errs []error
+	for _, configSecret := range configuration.Secrets {
+		if _, ok := deviceSecrets[configSecret.Name]; ok {
+			err = w.workloads.UpdateSecret(configSecret.Name, configSecret.Data)
```

Comment:
If I am not mistaken, there will be big load to podman secret creation/deleteion, because you re-create the secret each time the `updateSecret` is called, which would be every heartbeat. Could we somehow check if there is change and re-create the secret only if there is change and don't re-create the secret always?

Reply:
+1